File: ash2/ash/Service/Tb_userService.cs
using System;
using System.Linq;
using System.Web;
using model;
using Spring.Transaction.Interceptor;
using ash.Dao.Interf;
using System.Collections;

namespace ash.Service
{
    [Transaction]
    public class Tb_userService : ITb_userService
    {
        public ITb_userDao dao { set; get; }

        [Transaction]
        public void test()
        {
            Tb_user t1 = new Tb_user();
            t1.id = 123;
            t1.name = "zhuzhen";
            t1.pwd = "321";
            dao.save(t1);

            Tb_user t2 = new Tb_user();
            t2.id = 321;
            t2.name = "guolili";
            t2.pwd = "123";
            dao.save(t2);

            foreach (Tb_user model in dao.getAll())
            {
                string name = model.name;
            }
        }

        [Transaction(ReadOnly = true)]
        public Tb_user get(int id)
        {
            return dao.get(id);
        }

        [Transaction]
        public void save(Tb_user model)
        {
            dao.save(model);
        }

        [Transaction]
        public void update(Tb_user model)
        {
            dao.update(model);
        }

        [Transaction]
        public void delete(Tb_user model)
        {
            dao.delete(model);
        }

        [Transaction(ReadOnly = true)]
        public IList getAll()
        {
            return dao.getAll();
        }
    }
}

File: ssh/src/service/impl/Tb_userService.java
package service.impl;

import java.util.List;

import model.test.Tb_user;

import org.springframework.transaction.annotation.Transactional;

import service.interf.ITb_userService;
import dao.interf.ITb_userDao;

public class Tb_userService implements ITb_userService {
	private ITb_userDao dao;

	public ITb_userDao getDao() {
		return dao;
	}

	public void setDao(ITb_userDao dao) {
		this.dao = dao;
	}

	@Override
	@Transactional
	public void delete(Tb_user model) {
		dao.delete(model);
	}

	@Override
	@Transactional
	public void deleteById(int id) {
		dao.deleteById(id);
	}

	@Override
	@Transactional
	public Tb_user getById(int id) {
		return dao.getById(id);
	}

	@SuppressWarnings("unchecked")
	@Override
	@Transactional(readOnly = true)
	public List getAll() {
		return dao.getAll();
	}

	@Override
	@Transactional
	public void save(Tb_user model) {
		dao.save(model);
	}

	@Override
	@Transactional
	public void update(Tb_user model) {
		dao.update(model);
	}

}
